src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
Assuming the above comments, we should refactor this line to something like this:

```js
return averageDiff/children.length;
```

We don't need to check if averageDiff is a number, because the reduce method will return the initial value (in our case is 0) if the reducer function returns undefined.

It's not considered a good practice, but i recommend storing this calculation in a variable, to achieve a more readable code.